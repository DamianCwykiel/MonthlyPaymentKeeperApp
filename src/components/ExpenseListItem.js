import React from 'react';
import { Link } from 'react-router-dom';
import moment from 'moment';
import numeral from 'numeral';

//PLN currency
numeral.register('locale', 'pln', {
    delimiters: {
        thousands: ' ',
        decimal: ','
    },
    abbreviations: {
        thousand: 'k',
        million: 'm',
        billion: 'b',
        trillion: 't'
    },
    ordinal : function (number) {
        return number === 1 ? 'er' : 'ème';
    },
    currency: {
        symbol: 'zł'
    }
});

// switch between locales
numeral.locale('pln');

export const ExpenseListItem = ({ id, description, amount, createdAt }) => (
    <Link className="list-item" to = {`/edit/${id}`}>
        <div>
            <h3 className="list-item__title">{description}</h3>
            <h3 className="list-item__sub-title">{moment(createdAt).format('MMMM Do YYYY')}</h3>
        </div>
        <h3 className="list-item__data">
            <p>{numeral(amount / 100).format('0,0.00 $')}</p>
            <p>~ {numeral((amount / 100) / 4.30).format('0,0.00')} €.</p>
        </h3> 
    </Link>
); 
export default ExpenseListItem;