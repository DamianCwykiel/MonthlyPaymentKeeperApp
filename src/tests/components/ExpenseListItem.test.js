import React from 'react';
import { shallow } from 'enzyme';
import ExpenseListItem from '../../components/ExpenseListItem';
import expenses from '../fixtures/expenses'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

//Adapter configuration
Enzyme.configure({adapter: new Adapter()});

test('should render ExpenseListItem', () => {
    const wrapper = shallow(<ExpenseListItem {...expenses[0]}/>);
    expect(wrapper).toMatchSnapshot();
});

