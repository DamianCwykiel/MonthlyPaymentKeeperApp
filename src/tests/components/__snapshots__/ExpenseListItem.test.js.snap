// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpenseListItem 1`] = `
<Link
  className="list-item"
  replace={false}
  to="/edit/1"
>
  <div>
    <h3
      className="list-item__title"
    >
      Gum
    </h3>
    <h3
      className="list-item__sub-title"
    >
      January 1st 1970, 01:00:00
    </h3>
  </div>
  <h3
    className="list-item__data"
  >
    1,95 zł
    ~
    0,45
     €.
  </h3>
</Link>
`;
